Terminal.Gui/View/Layout/DimAuto.cs

```diff
@@ -94,17 +95,17 @@ internal override int Calculate (int location, int superviewContentSize, View us
                 // [ ] DimView
                 if (dimension == Dimension.Width)
                 {
-                    subviews = us.Subviews.Where (v => v.X is not PosAnchorEnd
-                                                       && v.X is not PosAlign
-                                                       && v.X is not PosCenter
-                                                       && v.Width is not DimFill).ToList ();
+                    subviews = includedSubviews.Where (v => v.X is not PosAnchorEnd
+                                                           && v.X is not PosAlign
+                                                           // && v.X is not PosCenter
+                                                           && v.Width is not DimFill).ToList ();
                 }
                 else
                 {
-                    subviews = us.Subviews.Where (v => v.Y is not PosAnchorEnd
-                                                       && v.Y is not PosAlign
-                                                       && v.Y is not PosCenter
-                                                       && v.Height is not DimFill).ToList ();
+                    subviews = includedSubviews.Where (v => v.Y is not PosAnchorEnd
```

Comment:
Not for this PR, but definitely needed for performance and resource usage at some point:

We should make things around collections way lighter than they are such as here.

The power of LINQ shines when it is able to defer evaluation til materializing is actually necessary.

Every time we call ToArray or ToList, the loop is evaluated, which means that, unconditionally, a new collection is constructed, allocated, and copied from the source, and that's VERY expensive.

Especially anything that happens in the render pipeline should be as lightweight as possible, as that code is all very hot.

There are various strategies to help with this, and no single strategy is appropriate for all uses.

Some examples:
 - Using `ArraySegment<T>`s when contiguous ranges of a source array are needed.
 - Using `Memory<T>` or `Span<T>` over collections, as and when appropriate for each of those.
   - Using slices, when using either of those, for pretty much the same reason one would use `ArraySegment<T>` without them.
 - Never calling .ToList() or similar until the very last moment, and only when that's because you now both need to enumerate the data and then operate on the subset ***and*** are also reasonably certain that multiple enumeration would happen if you don't do it, ***and*** that that multiple enumeration is actually more expensive than allocating a new collection (the cutoff for that is situational but often pretty early).